### PR TITLE
adding more docs

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Routing/Matching/IEndpointComparerPolicy.cs [hunk 1]
```diff
@@ -5,8 +5,30 @@
 
 namespace Microsoft.AspNetCore.Routing.Matching
 {
+    /// <summary>
+    /// A <see cref="MatcherPolicy"/> interface that can implemented to sort
```
Comment (kichalla) [inline src/Microsoft.AspNetCore.Routing/Matching/IEndpointComparerPolicy.cs]: be implemented

File: src/Microsoft.AspNetCore.Routing/Matching/IEndpointComparerPolicy.cs [hunk 2]
```diff
@@ -5,8 +5,30 @@
...
+    /// the dependency injection container as singleton services of type <see cref="MatcherPolicy"/>.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Candidates in a <see cref="CandidateSet"/> are sorted based on their priority. Defining
+    /// a <see cref="IEndpointComparerPolicy"/> adds an additional criterion to the sorting
+    /// operation used to order candidates.
+    /// </para>
+    /// <para>
+    /// As an example, the implementation of <see cref="HttpMethodMatcherPolicy"/> implements
```
Comment (kichalla) [inline src/Microsoft.AspNetCore.Routing/Matching/IEndpointComparerPolicy.cs]: 👍

File: src/Microsoft.AspNetCore.Routing/Matching/MatcherEndpoint.cs [hunk 3]
```diff
@@ -9,13 +9,29 @@
 
 namespace Microsoft.AspNetCore.Routing.Matching
 {
+    /// <summary>
+    /// Represents an <see cref="Endpoint"/> that can be used in URL matching.
```
Comment (kichalla) [inline src/Microsoft.AspNetCore.Routing/Matching/MatcherEndpoint.cs]: URL matching or URL generation?

File: src/Microsoft.AspNetCore.Routing/Matching/CandidateSet.cs [hunk 4]
```diff
@@ -18,8 +23,13 @@ public sealed class CandidateSet
 
         private CandidateState[] _additionalCandidates;
 
-        // Provided to make testing possible/easy for someone implementing
-        // an EndpointSelector.
+        /// <summary>
+        /// Creates a new candidate set with the provided list of endpoints and associated scores.
```
Comment (JamesNK) [inline src/Microsoft.AspNetCore.Routing/Matching/CandidateSet.cs]: nit: Initializes a new instance of the `CandidateSet` structure with the provided list of endpoints and associated scores.

File: src/Microsoft.AspNetCore.Routing/Matching/EndpointMetadataComparer.cs [hunk 5]
```diff
@@ -6,10 +6,30 @@
...
+    /// <summary>
+    /// A base class for <see cref="IComparer{Endpoint}"/> implementations that use 
+    /// a specific type of metadata from <see cref="Endpoint.Metadata"/> for comparison.
+    /// Useful for implementing <see cref="IEndpointComparerPolicy.Comparer"/>.
+    /// </summary>
+    /// <typeparam name="TMetadata">
+    /// The type of metadata to compare. Typically this is a type of metadata related
+    /// to the application concern being handled.
+    /// </typeparam>
     public abstract class EndpointMetadataComparer<TMetadata> : IComparer<Endpoint> where TMetadata : class
```
Comment (JamesNK) [inline src/Microsoft.AspNetCore.Routing/Matching/EndpointMetadataComparer.cs]: Off topic: Should this class be suffixed with "Base"?

File: src/Microsoft.AspNetCore.Routing/Matching/MatcherEndpoint.cs [hunk 6]
```diff
@@ -9,13 +9,29 @@
...
+    /// </summary>
     public sealed class MatcherEndpoint : Endpoint
     {
         internal static readonly Func<RequestDelegate, RequestDelegate> EmptyInvoker = (next) =>
         {
             return (context) => Task.CompletedTask;
         };
 
+        /// <summary>
+        /// Creates a new <see cref="MatcherEndpoint"/>.
```
Comment (JamesNK) [inline src/Microsoft.AspNetCore.Routing/Matching/MatcherEndpoint.cs]: Initializes a new instance of the `MatcherEndpoint` class.

**Pull request merged**